// fetchye-core/lib/actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.loadingAction = loadingAction;
exports.setAction = setAction;
exports.deleteAction = deleteAction;
exports.errorAction = errorAction;
exports.clearErrorsAction = clearErrorsAction;

var _constants = require("./constants");

/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
function loadingAction({
  hash
}) {
  return {
    type: _constants.IS_LOADING,
    hash
  };
}

function setAction({
  hash,
  value
}) {
  return {
    type: _constants.SET_DATA,
    hash,
    value
  };
}

function deleteAction({
  hash
}) {
  return {
    type: _constants.DELETE_DATA,
    hash
  };
}

function errorAction({
  hash,
  error
}) {
  return {
    type: _constants.ERROR,
    hash,
    error
  };
}

function clearErrorsAction({
  hash
}) {
  return {
    type: _constants.CLEAR_ERROR,
    hash
  };
}
