// fetchye-core/lib/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CLEAR_ERROR = exports.ERROR = exports.DELETE_DATA = exports.SET_DATA = exports.IS_LOADING = exports.ACTION_NAMESPACE = void 0;

/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
const ACTION_NAMESPACE = '@fetchye';
exports.ACTION_NAMESPACE = ACTION_NAMESPACE;
const IS_LOADING = `${ACTION_NAMESPACE}/IS_LOADING`;
exports.IS_LOADING = IS_LOADING;
const SET_DATA = `${ACTION_NAMESPACE}/SET_DATA`;
exports.SET_DATA = SET_DATA;
const DELETE_DATA = `${ACTION_NAMESPACE}/DELETE_DATA`;
exports.DELETE_DATA = DELETE_DATA;
const ERROR = `${ACTION_NAMESPACE}/ERROR`;
exports.ERROR = ERROR;
const CLEAR_ERROR = `${ACTION_NAMESPACE}/CLEAR_ERROR`;
exports.CLEAR_ERROR = CLEAR_ERROR;
